src/Pim/Component/Catalog/Manager/ProductUniqueDataSynchronizer.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Pim\Component\Catalog\Manager;
```

Comment:
Quite weird for me to put it in the Manager folder (that i would :fire: at some point :) ).

I would put it close to the Saver (same folder) which is the only class to use it to ensure the sync (it may be part of this Saver in fact). Other option would be to put it close to the product model (more DDD oriented but not consistent with our current stack and not making to much sense maybe as it's not ensure by the model by itself).